### PR TITLE
ADD JAR command wih Session

### PR DESCRIPTION
**Pull request by jjlopezm**
## Description
This PR enables again the commands ADD JAR, ADD APP and EXECUTE, now with the Session implementation. 
### Testing
- [ ] Unit, integration tests

### Documentation
- [ ] Changelog update
- [ ] Documentation link

### CODE REVIEW

File: server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala [hunk 1]
```diff
@@ -45,41 +46,43 @@ class CrossdataHttpServer(config: Config, serverActor: ActorRef, implicit val sy
...
-              val hdfsConfig = XDContext.xdConfig.getConfig("hdfs")
-              //Send a broadcast message to all servers
-              val hdfsPath = writeJarToHdfs(hdfsConfig, path)
-              mediator ! Publish(AddJarTopic, CommandEnvelope(AddJARCommand(hdfsPath, hdfsConfig = Option(hdfsConfig)), new Session(UUID.randomUUID(), serverActor)))
-              hdfsPath
-            }
+        }.runFold(Map.empty[String, Any])((map, tuple) => map + tuple)
+
+        val done = allPartsF.onSuccess {
+          case _ => logger.info("Recieved file")
```
Comment (miguel0afd) [inline server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala]: It looks like value done is not used in any place. In addition, maybe we could take advantage of the "onSuccess(allPartsF)" in order to write in the log and remove this part

**Comment by miguel0afd** [comment]
LGTM

File: driver/src/main/scala/com/stratio/crossdata/driver/util/HttpClient.scala [hunk 2]
```diff
@@ -47,12 +50,14 @@ class HttpClient(ctx: HttpClientContext) {
...
   private val http = Http(actorSystem)
 
-  def sendJarToHTTPServer(path: String): Future[String] = {
+  def sendJarToHTTPServer(path: String, session: Session): Future[String] = {
     val host = config.getCrossdataServerHost.split(':').head
-    for(
-      request <- createRequest(s"http://$host:13422/upload", new File(path));
+    val sessionUUID = session.id
+    //val sessionActorRef = Serialization.serializedActorPath(session.clientRef)
+    for (
```
Comment (pianista215) [inline driver/src/main/scala/com/stratio/crossdata/driver/util/HttpClient.scala]: Better put port into in a val ?

File: server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala [hunk 3]
```diff
@@ -45,41 +46,41 @@ class CrossdataHttpServer(config: Config, serverActor: ActorRef, implicit val sy
...
-          }
+    path("upload" / JavaUUID) { sessionUUID =>
+      entity(as[Multipart.FormData]) { formData =>
+        // collect all parts of the multipart as it arrives into a map
+        var path = ""
+        val allPartsF: Future[Map[String, Any]] = formData.parts.mapAsync[(String, Any)](1) {
+
+          case part: BodyPart if part.name == "fileChunk" =>
+            // stream into a file as the chunks of it arrives and return a future file to where it got stored
+            val file = new java.io.File(s"/tmp/${part.filename.getOrElse("uploadFile")}")
```
Comment (pianista215) [inline server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala]: Instead of leaving in /tmp the file, delete it after processed it.

**Comment by pianista215** [comment]
:+1:

**Pull request merged**